snapcraft/plugins/nodejs.py

```diff
@@ -232,6 +242,9 @@ def _install(self, rootdir):
                 os.path.join(package_dir, "yarn.lock"),
             )
 
+        flags.append("--offline")
```

Comment:
What's the reason for '--offline' here?
Seems that it breaks my build:

> 
> /root/parts/server/npm/bin/npm install --unsafe-perm --offline --prod
> npm ERR! code ENOTCACHED
> npm ERR! request to https://registry.npmjs.org/@feathersjs%2fexpress failed: cache mode is 'only-if-cached' but no cached response available.
>

Reply:
The previous code
```
        if rootdir == self.builddir:
            flags = ["--offline", "--prod"]
```
actually was never executed because it was only called `self._install(rootdir=self.sourcedir)`

Reply:
Workaround was to implicitly install globally all packages with preinstall hook i.e.

```json
  "scripts": {
    "preinstall": "npm install --global @feathersjs/express express socket.io avahi-dbus github:Andersbakken/node-posix#node12fix"
  },  
```

which in its own turn failed (because `SUDO_UID` fix doesn't cower `github:...` modules) with this error:

> \> server@1.0.0 preinstall /root/parts/server/build
> \> npm install --global @feathersjs/express express socket.io avahi-dbus github:Andersbakken/node-posix#node12fix
> 
> npm ERR! Error while executing:
> npm ERR! /usr/bin/git ls-remote -h -t ssh://git@github.com/Andersbakken/node-posix.git
> npm ERR! 
> npm ERR! fatal: failed to stat '/root/parts/server/build': Permission denied
> npm ERR! 
> npm ERR! exited with error code: 128

which remind me that [already fixed that issue few months ago](https://forum.snapcraft.io/t/multiple-problems-when-build-on-mac/12693/5?u=palavrov) - the proper fix is to set these environment variables in `snapcraft.yaml`

```yaml
        build-environment:
            - SUDO_UID: '0'
            - SUDO_GID: '0'
            - SUDO_USER: 'root'
```

and finally my build was able to succeed.